src/main/java/com/ibm/watson/developer_cloud/spring/boot/WatsonAutoConfiguration.java

```diff
@@ -215,7 +248,9 @@ public ToneAnalyzer toneAnalyzer() {
   public VisualRecognition visualRecognition() {
     VisualRecognition service = new VisualRecognition(vrConfig.getVersionDate());
     configUrl(service, vrConfig);
-    configBasicAuth(service, vrConfig);
+    if (!configBasicAuth(service, vrConfig)) {
```

Comment:
Does this line need to check for `configApiKey()`?

Reply:
Good catch.  I think it does.  And actually I think I will flip all these tests around to check for IAM first, and only when it is not provided to use the "legacy" auth.  That will be a smaller code change.